app/models/mixins/amazon_s3_authentication_mixin.rb

```diff
@@ -0,0 +1,86 @@
+module AmazonS3AuthenticationMixin
+  extend ActiveSupport::Concern
+
+  #
+  # Connections
+  #
+  def connect(options = {})
+    username = options[:username] || authentication_userid(options[:auth_type])
+    password = options[:password] || authentication_password(options[:auth_type])
+    service  = :S3
+    # Note: The hard-coded region will be removed after manageiq-ui-class implements region selection
+    region   = options[:region] || "us-east-1"
+
+    self.class.raw_connect(username, password, service, region)
+  end
+
+  def with_depot_connection(options = {})
+    raise _("no block given") unless block_given?
+    _log.info("Connecting through #{self.class.name}: [#{name}]")
+    yield connect(options)
+  end
+
+  def verify_credentials(auth_type = nil, options = {})
+    _log.debug("verifying credentials for auth_type [#{auth_type}] with options [#{options}]")
+
+    self.class.connection_rescue_block do
+      # EC2 does Lazy Connections, so call a cheap function
+      with_depot_connection(options.merge(:auth_type => auth_type)) do |ec2|
+        self.class.validate_connection(ec2)
+      end
+    end
+
+    true
+  end
+
+  module ClassMethods
```

Comment:
Why did you choose to implement these as class methods? I'm not sure I see the advantage.

Reply:
Again - refer to the previous comment - pulled out of the AWS provider... I didn't give it *that* much original thought - the authentication was already working for the provider....